pkg/cmd/server/origin/master_config.go

```diff
@@ -145,6 +146,14 @@ func BuildMasterConfig(options configapi.MasterConfig) (*MasterConfig, error) {
 	admissionControlPluginNames := []string{"OriginNamespaceLifecycle"}
 	admissionController := admission.NewFromPlugins(privilegedLoopbackKubeClient, admissionControlPluginNames, "")
 
+	// This plugin must be manually initialized, it requires an openshift client which is not
+	// available via the factory interface normally used in NewFromPlugins.
+	routeAdmissionPlugin, err := routeadmission.NewRouteAdmissionPlugin(privilegedLoopbackKubeClient, privilegedLoopbackOpenShiftClient)
+	if err != nil {
+		return nil, err
+	}
+	admissionController = admission.NewChainHandler(admissionController, routeAdmissionPlugin)
```

Comment:
Ok, cool.  This was my main concern.  Thanks for the other comments too, will fix if we move forward with this PR.